llama-index-core/llama_index/core/readers/base.py

```diff
@@ -25,10 +25,21 @@ def lazy_load_data(self, *args: Any, **load_kwargs: Any) -> Iterable[Document]:
             f"{self.__class__.__name__} does not provide lazy_load_data method currently"
         )
 
+    async def alazy_load_data(
+        self, *args: Any, **load_kwargs: Any
+    ) -> Iterable[Document]:
+        """Load data from the input directory lazily."""
+        # Fake async - just calls the sync method. Override in subclasses for real async implementations.
+        return self.lazy_load_data(*args, **load_kwargs)
+
     def load_data(self, *args: Any, **load_kwargs: Any) -> List[Document]:
         """Load data from the input directory."""
         return list(self.lazy_load_data(*args, **load_kwargs))
 
+    async def aload_data(self, *args: Any, **load_kwargs: Any) -> List[Document]:
+        """Load data from the input directory."""
+        return self.load_data(*args, **load_kwargs)
```

Comment:
should this call `alazy_load_data()` ?

Reply:
Not sure it really matters I guess

Reply:
I think it should prob be calling `load_data` actually, because I don't think all the reader implementations actually implement a `lazy_load_data`, they just implement `load_data`

Reply:
ah gotcha